deploy/operator/provision-shared-host.sh

```diff
@@ -0,0 +1,146 @@
+#!/bin/bash
+set -o verbose
+set -eu
+set -o pipefail
+
+# --- Error Handling Function ---
+handle_error() {
+    local EXIT_CODE=$?
+    local FAILED_COMMAND=$BASH_COMMAND
+
+    printf "{message: \"'%s' failed with exit code %i.\", level: \"ERROR\"}" "$FAILED_COMMAND" "$EXIT_CODE" >&2
+}
+# Trap the ERR signal to run the function
+trap 'handle_error' ERR
+# ------------------------------
+
+trap 'ssh -O exit $SSH_HOST 2>/dev/null || true' EXIT # end SSH session upon EXIT
+
+cd /tmp
+mkdir -p /root/.ssh
+cp "$SSH_WORKSPACE_PATH/id_rsa" /tmp/master_key
+chmod 0400 /tmp/master_key
+export SSH_HOST="$USER@$HOST"
+SSH_MULTIPLEX_OPTS=(-o ControlMaster=auto -o ControlPath=/tmp/ssh-%r@%h:%p)
+SSH_OPTS=(-i /tmp/master_key -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null  "${SSH_MULTIPLEX_OPTS[@]}")
+
+USERNAME=u-$(echo "$TASKRUN_NAME$NAMESPACE" | md5sum | cut -b-28)
+export USERNAME
+
+# Create script to provision VM
+cat >script.sh <<EOF
+sudo dnf install podman -y
+
+rm -f "$USERNAME" "$USERNAME".pub
+
+# Retry useradd command as sometimes it hits a "cannot lock /etc/passwd; try again later" error
+for i in {10..1}; do
+  sudo useradd -m "$USERNAME" -p \$(openssl rand -base64 12)
+  if [ \$? -eq 0 ]; then
+    echo "{message: \"User $USERNAME added successfully.\", level: \"INFO\"}"
+    break
+  fi
+  ATTEMPTS_REMAINING=\$((i - 1))
+  if [ "\$ATTEMPTS_REMAINING" -gt 0 ]; then
+    echo "{message: \"User $USERNAME addition attempt failed, retrying \$ATTEMPTS_REMAINING more times...\", level: \"WARNING\"}"
+    sleep 1
+  else
+    echo "{message: \"Failed to add user $USERNAME after 10 total attempts.\", level: \"ERROR\"}"
+    exit 1
+  fi
+done
+
+ssh-keygen -N '' -f "$USERNAME"
+sudo su "$USERNAME" -c 'mkdir /home/"$USERNAME"/.ssh'
+sudo su "$USERNAME" -c 'mkdir /home/"$USERNAME"/build'
```

Comment:
Just realized: if we use single quotes for the command here, what will be executed remotely will be
```
mkdir /home/"$USERNAME"/.ssh
```
as the variable wont' be expanded inside single quotes.
if we want this to work as expected, it is better to surround the command with double quotes so the variable is expanded:
```
"mkdir /home/$USERNAME/.ssh"
```